### PR TITLE
Implement scoring matrix downloading process

### PR DESCRIPTION
**Pull request by clonder**
Automatic Downloading: Matrix files are automatically downloaded if missing, ensuring up-to-date resources.
Caching: Downloaded files are cached locally, reducing redundant downloads and improving performance.

### CODE REVIEW

**Comment by Accio** [comment]
Thanks @clonder! I will have a look.

**Comment by alexandra-valeanu** [comment]
Hey, @clonder! Thank you for working on Kinex!

I tested the changes you made and encountered some errors:
IMG_7947

Consider adding "requests" to  dependencies in `pyproject.toml`

After I install the "requests" package I get the following:
IMG_5138

Here is a link to how the [project.urls] are used: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls
Consider adding the links to `resources` or store them in variables.

**Comment by clonder** [comment]
Hi @alexandra-valeanu,

Thank you for reviewing this package. I’ve added resources to the dependencies—sorry for the inconvenience.

I was initially testing this within the same directory, so the `pyproject.toml` was always in the path. My plan was to include it in the build when publishing to PyPI in the next iteration.

We can use resources also, I’ve updated it to use the configuration files as per this commit.

Thanks again for your feedback!

**Comment by alexandra-valeanu** [comment]
Hey @clonder! All good, it works now. Could you change the documentation in `docs/chapters/features/usage.rst`?

**Comment by clonder** [comment]
Sure!

File: =0.12. [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Collecting esbonio
+  Downloading esbonio-0.16.5-py3-none-any.whl.metadata (4.7 kB)
+Collecting sphinx (from esbonio)
+  Downloading sphinx-8.1.3-py3-none-any.whl.metadata (6.4 kB)
+Requirement already satisfied: platformdirs in /home/clony/v/lib/python3.12/site-packages (from esbonio) (4.3.6)
+Collecting pygls<2,>=1.1.0 (from esbonio)
+  Downloading pygls-1.3.1-py3-none-any.whl.metadata (4.7 kB)
```
Comment (alexandra-valeanu) [inline =0.12.]: @clonder was this file added by mistake?

Comment (clonder) [reply =0.12.]: Yes

**Review by alexandra-valeanu — APPROVED** [review]

**Comment by Accio** [comment]
thank you both!

**Pull request merged**